File: nodesms/controllers/courseController.js
const courseService = require('../services/courseService');

exports.createCourse = async (req, res) => {
  try {
    const courseDto = await courseService.createCourse(req.body);
    res.status(201).json(courseDto);
  } catch (error) {
    res.status(500).json({ message: error.message });
  }
};

exports.getAllCourses = async (req, res) => {
  try {
    const courses = await courseService.getAllCourses();
    res.json(courses);
  } catch (error) {
    res.status(500).json({ message: error.message });
  }
};

exports.getCourseById = async (req, res) => {
  try {
    const course = await courseService.getCourseById(req.params.id);
    res.json(course);
  } catch (error) {
    res.status(404).json({ message: error.message });
  }
};

File: student-management/src/main/java/com/example/student_management/controller/CourseController.java
package com.example.student_management.controller;


import com.example.student_management.dto.CourseDto;

import com.example.student_management.dto.CreateCourseDto;

import com.example.student_management.services.CourseService;

import org.springframework.http.ResponseEntity;

import org.springframework.web.bind.annotation.*;


import java.util.List;


@RestController
@RequestMapping("/courses")
public class CourseController {

    private final CourseService courseService;

    public CourseController(CourseService courseService) {
        this.courseService = courseService;

    }

    @PostMapping("/create")
    public ResponseEntity<CourseDto> createCourse(@RequestBody CreateCourseDto createCourseDto) {
        return ResponseEntity.ok(courseService.createCourse(createCourseDto));

    }

    @GetMapping
    public ResponseEntity<List<CourseDto>> getAllCourses() {
        return ResponseEntity.ok(courseService.getAllCourses());

    }

    @GetMapping("/{id}")
    public ResponseEntity<CourseDto> getCourseById(@PathVariable Long id) {
        return ResponseEntity.ok(courseService.getCourseById(id));

    }
}
